src/main/java/com/heeverse/config/MultiDataSource.java

```diff
@@ -0,0 +1,71 @@
+package com.heeverse.config;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.core.env.Environment;
+import org.springframework.util.Assert;
+
+import java.util.Arrays;
+
+import static com.heeverse.common.Constants.*;
+
+/**
+ * @author jeongheekim
+ * @date 10/31/23
+ */
+@Configuration
+public class MultiDataSource {
+    private final VaultOperationService vaultOperationService;
+    private final Environment environment;
+
+    public MultiDataSource(VaultOperationService vaultOperationService, Environment environment) {
+        this.vaultOperationService = vaultOperationService;
+        this.environment = environment;
+    }
+
+    @Primary
+    @Bean(name = "primaryDataSource")
+    public HikariDataSource primaryDataSource() {
+        return new HikariDataSource(getDataSourceProperties());
+    }
+
+    @Bean(name = "lockDataSource")
+    public HikariDataSource lockDataSource() {
+        return new HikariDataSource(getDataSourceProperties());
+    }
+
+    private HikariConfig getDataSourceProperties() {
+        String[] activeProfiles = environment.getActiveProfiles();
+        HikariConfig hikariConfig = new HikariConfig();
+        if (Arrays.asList(activeProfiles).contains(LOCAL)) {
+            hikariConfig.setJdbcUrl(environment.getProperty("spring.datasource.url"));
+            hikariConfig.setDriverClassName(environment.getProperty("spring.datasource.driver-class-name"));
+            hikariConfig.setUsername(environment.getProperty("spring.datasource.username"));
+        } else {
+            var dbProps = vaultOperationService.getProps(VAULT_PATH, MYSQL_SECRETES, DBProps.class);
+            hikariConfig.setJdbcUrl(dbProps.url());
+            hikariConfig.setUsername(dbProps.username());
+            hikariConfig.setPassword(dbProps.password());
+        }
```

Comment:
중복된 코드는 이렇게 제거할 수 있습니다. 코드 중복을 피해주세요.
```suggestion
        DBProps dbProps = getDBPropsBy(profiles);
        hikariConfig.setJdbcUrl(dbProps.url());
        hikariConfig.setUsername(dbProps.username());
        hikariConfig.setPassword(dbProps.password());
        return hikariConfig;
```